handlers.go

```diff
@@ -3,14 +3,16 @@ package main
 import (
 	"encoding/json"
 	"fmt"
+	"github.com/gorilla/mux"
 	"net/http"
+	"strconv"
 )
 
 type DeviceHandlers struct {
 	service *Service
 }
 
-func (h *DeviceHandlers) devicesHandler(w http.ResponseWriter, r *http.Request) {
+func (dh *DeviceHandlers) addDeviceHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
I will encourage you to mak those `http.Handler` methods public. The reason is that `DeviceHandlers` only purpose is to provide handlers - it is part of its public contract.